api/runtime_env.proto

```diff
@@ -0,0 +1,264 @@
+// Copyright 2017 The OpenPitrix Authors. All rights reserved.
+// Use of this source code is governed by a Apache license
+// that can be found in the LICENSE file.
+
+
+syntax = "proto3";
+package openpitrix;
+
+// set go package name to pb
+option go_package = "pb";
+
+import "google/protobuf/wrappers.proto";
+import "google/protobuf/timestamp.proto";
+import "google/api/annotations.proto";
+import "protoc-gen-swagger/options/annotations.proto";
+
+option (grpc.gateway.protoc_gen_swagger.options.openapiv2_swagger) = {
+	info: {
+		title: "OpenPireix Project";
+		version: "0.0.1";
+		contact: {
+			name: "OpenPireix Project";
+			url: "https://openpitrix.io";
+		};
+	};
+	schemes: HTTP;
+	consumes: "application/json";
+	produces: "application/json";
+};
+
+message RuntimeEnv{
+	google.protobuf.StringValue runtime_env_id = 1;
+	google.protobuf.StringValue name = 2;
+	google.protobuf.StringValue description = 3;
+	google.protobuf.StringValue runtime_env_url = 4;
+	google.protobuf.StringValue runtime_env_credential_id = 5;
+	google.protobuf.StringValue labels = 6;
+	google.protobuf.StringValue owner = 7;
+	google.protobuf.StringValue status = 8;
+	google.protobuf.Timestamp create_time = 9;
+	google.protobuf.Timestamp status_time = 10;
+}
+
+message CreateRuntimeEnvRequest {
+	google.protobuf.StringValue name = 1;
+	google.protobuf.StringValue description = 2;
+	google.protobuf.StringValue labels = 3;
+	google.protobuf.StringValue runtime_env_url = 4;
+}
+
+message CreateRuntimeEnvResponse{
+	RuntimeEnv runtime_env = 1;
+}
+
+message DescribeRuntimeEnvsRequest{
+	repeated google.protobuf.StringValue runtime_env_ids = 1;
```

Comment:
`runtime_env_ids`, you can replace it with `runtime_env_id`.

Reply:
For repeated types of data, should we use the plural form of the word

Reply:
In protobuf's offical doc, they use the plural form of words.

Reply:
In a describe requests, `repeated` filter use single number. This approach references most cloud vendors.